icu4j/main/core/src/main/java/com/ibm/icu/message2/MFParser.java

```diff
@@ -46,22 +46,38 @@ public static MFDataModel.Message parse(String input) throws MFParseException {
     // Parser proper
     private MFDataModel.Message parseImpl() throws MFParseException {
         MFDataModel.Message result;
+        // Determine if message is simple or complex; this requires
+        // looking through whitespace.
+        int savedPosition = input.getPosition();
+        while (true) {
```

Comment:
Can we use skipOptionalWhitespaces() for this loop?

Reply:
Not straightforwardly, because if the message is simple, the whitespace can't be skipped (has to be kept as part of the message text). To me it seemed simplest to write code that just looks ahead without consuming any input.

Reply:
Done in [9bd9556](https://github.com/unicode-org/icu/pull/3063/commits/9bd9556025ad236bb0c94b7954b0efe857f4b73a)